rpc/baremetalhostagent/agent/moc_baremetalhostagent.proto

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache v2.0 license.
+
+syntax = "proto3";
+option go_package = "github.com/microsoft/moc/rpc/baremetalhostagent";
+
+package moc.baremetalhostagent;
+import "google/protobuf/wrappers.proto";
+import "moc_common_common.proto";
+import "moc_common_computecommon.proto";
+
+
+message BareMetalHostSSHPublicKey {
+	string keydata = 1;
+}
+
+message BareMetalHostDisk {
+	string diskName = 1;
+	int32 diskSizeGB = 2;
+}
+
+message BareMetalHostStorageConfiguration {
+	repeated BareMetalHostDisk disks = 1;
+	string imageReference = 2;
+}
+
+message BareMetalHostUserConfiguration {
+	string username = 1;
+	string password = 2;
+	UserType usertype = 3;
+}
+
+message BareMetalHostRDPConfiguration {
+	bool disableRDP = 1;
+}
+
+message BareMetalHostLinuxConfiguration {
+	bool disablePasswordAuthentication = 1;
+}
+
+message BareMetalHostWindowsConfiguration {
+	bool enableAutomaticUpdates = 1;
+	string timeZone = 2;
+	BareMetalHostRDPConfiguration RDPConfiguration = 3;
+}
+
+message BareMetalHostOperatingSystemConfiguration {
+	string computerName = 1;
+	BareMetalHostUserConfiguration administrator = 2;
+	repeated BareMetalHostUserConfiguration users = 3;
+	string customData = 4;
```

Comment:
I thought the only feature that we need is customData, which has the cloudinit - which could potentially do everything
1. User creation
2. ssh keys 
3. hostname setting
What is the intent is exposing all these properties that cloudagent has to fill and push to this agent

Reply:
Presumably these were mirrors from the VM entity. I believe the logic to generate cloud init was in node agent, not the cloud agent, and so that is probably why these are passed down directly (but you know more here).

Reply:
These are mirrors from VM entity to create the provisioning iso drive. I keep the logic to generate cloud-init drive the same as with Compute/VM so I am keeping this struct the same. Depends on the demo condition (whether cloudagent would use only customData, or all), I will revisit this and remove if needed.

Reply:
Remove all of these for now since we only need cloud-init script (customData)